apps/studio/src/backend/lib/OfflineLicense.ts

```diff
@@ -0,0 +1,131 @@
+
+import platformInfo from '@/common/platform_info'
+import path from 'path'
+import crypto from 'crypto'
+import { existsSync, readFileSync } from 'fs'
+import { keysToStatus, LicenseKey } from '@/common/appdb/models/LicenseKey'
+import { LicenseStatus } from '@/lib/license'
+import rawLog from '@bksLogger'
+import { snakeCaseObjectKeys } from '@/common/utils'
+import { TransportLicenseKey } from '@/common/transport'
+
+const log = rawLog.scope('OfflineLicense')
+
+let _cachedLicense = null
+
+/**
+ * Payload looks like this (for good license):
+ * {
+  "license_key": {
+    "valid_until": "2099-01-01T06:00:00.000Z",
+    "support_until": "2099-01-01T06:00:00.000Z",
+    "customer_support_until": "2099-01-01T06:00:00.000Z",
+    "created_at": "2024-12-14T03:38:52.989Z",
+    "license_type": "PersonalLicense",
+    "email": "123",
+    "id": 4,
+    "key": "e605ca57-2115-4b21-b1ec-690f997df38e",
+    "subscription": true,
+    "max_allowed_app_release": null
+  },
+  "errors": null,
+  "status": 200
+}
+ */
+
+
+export interface LicenseOptions {
+  licensePath?: string
+  keyPath?: string
+}
+
+export class OfflineLicense {
```

Comment:
Should this be in the commercial folder?

Reply:
Good question, but I think it's fine -- you can only export business licenses anyway, so no way they can use it without a business license.